core/account/utxodb/reserve.go

```diff
@@ -57,15 +58,23 @@ type (
 	}
 )
 
+func beginTxn(ctx context.Context) (pg.Tx, error) {
+	db := pg.FromContext(ctx)
+	beginner, ok := db.(pg.Beginner)
```

Comment:
I think pg.Beginner isn't the interface we need here. A real sql.DB has a different signature. See the deleted pg.begin:

```
type beginner interface {		
	Begin(context.Context) (*sql.Tx, error)		
}		
switch d := db.(type) {		
case beginner: // e.g. *sql.DB		
	return d.Begin(ctx)		
case Beginner: // e.g. pgtest.noCommitDB		
	return d.Begin(ctx)		
}
```